streams/shared/src/main/scala/zio/stream/StreamEffectChunk.scala

```diff
@@ -90,6 +90,20 @@ private[stream] class StreamEffectChunk[-R, +E, +A](override val chunks: StreamE
   override def mapConcatChunk[B](f: A => Chunk[B]): StreamEffectChunk[R, E, B] =
     StreamEffectChunk(chunks.map(_.flatMap(f)))
 
+  final def processChunk: ZManaged[R, E, () => A] =
+    chunks.processEffect.map { thunk =>
+      var counter         = 0
+      var chunk: Chunk[A] = Chunk.empty
+      def pull(): A = {
+        while (counter >= chunk.length) {
```

Comment:
```suggestion
        if (counter >= chunk.length) {
```

Reply:
this case doesn't cover case of an empty chunk! the same logic is used in the analogous method `ZStreamChunk.process`

Reply:
Oh right! Good catch!